### PR TITLE
Refactor permissions and add set account detail in genesis block

### PR DESCRIPTION
**Pull request by neewy**
### Description of the Change

This pull requests brings prettified file permissions.hpp, refactored order in primitive.proto and addition of domain-related permissions to the primitive.proto and pb_command_factory. The order is persistent across modified files, so it is convenient now to check what is the status of permissions.

### Benefits

Now genesis block has almost all possible permissions in it. The code looks a bit better.

### Possible Drawbacks 

1. Genesis block was modified manually. CLI code has to be modified as well.
2. Grant permission for TransferAsset is not checked.

### Usage Examples or Tests *[optional]*

pb_commands_tests.cpp
CommandTest, create_role

I also executed a manual test against modified genesis.block, and it works.

### CODE REVIEW

**Comment by neewy** [comment]
For now, if you add such permissions directly to the genesis block the daemon will not launch

File: irohad/model/permissions.hpp [hunk 1]
```diff
@@ -24,122 +24,186 @@
...
+    /*                           Domain                           */
+    const std::string can_create_domain = "can_create_domain";
+
+    /* ~~~~~~~~       Query-related permissions        ~~~~~~~~   */
+
+    //  The set of permissions below refer to the specific queries.
+    //  During stateful validations, these permissions are checked
+    //  to be assigned to query creator.
+    //  These permissions are divided into three groups:
+    // * my — query creator can only query its data
```
Comment (grimadas) [inline irohad/model/permissions.hpp]: It think 'only' is redundant here

File: schema/primitive.proto [hunk 2]
```diff
@@ -1,72 +1,103 @@
...
+ */
+
 syntax = "proto3";
+
+
 package iroha.protocol;
 
+
+/**
+ * Represents any possible value for permission field,
```
Comment (grimadas) [inline schema/primitive.proto]: ❤️

File: schema/primitive.proto [hunk 3]
```diff
@@ -1,72 +1,103 @@
...
+ * then the account is capable of performing such action.
+ *
+ * Command and Query related permissions are included in the role,
+ * which an Account can have.
+ *
+ * Grantable permission is being set as a relation between two accounts:
```
Comment (grimadas) [inline schema/primitive.proto]: 'is being set' - seem to be overcomplicated phrase

**Pull request merged**